### PR TITLE
media-tv/plex-media-server: 1.61.4 bump

### PR DESCRIPTION
**Pull request by ss23**
Closes: https://bugs.gentoo.org/681858

I'm taking over ownership of this package from Tony Vroon, under the proxy-maint project, with assistance from @juippis.

As this is just a bump that I've tested and tweaked from b.g.o, I haven't fixed any other issues within the project. Once I push the latest version of the ebuild (as even this version is quite out of date), I'll concentrate on fixing the outstanding bugs and making further changes to tidy it up.

### CODE REVIEW

File: acct-group/plex/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<!-- maintainer-needed -->
```
Comment (juippis) [inline acct-group/plex/metadata.xml]: We don't add new packages without maintainer to the tree.

File: acct-user/plex/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<!-- maintainer-needed -->
```
Comment (juippis) [inline acct-user/plex/metadata.xml]: Same here.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,157 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 )
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: (We are currently trying to get rid of python2-only software, you should REALLY start looking into getting python3 support for this, but it's not an issue now)

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+MINOR_VERSION="1469-6d5612c2f"
+
+_APPNAME="plexmediaserver"
+_USERNAME="plex"
+_SHORTNAME="${_USERNAME}"
+_FULL_VERSION="${PV}.${MINOR_VERSION}"
+
+URI="https://downloads.plex.tv/plex-media-server-new"
+
+DESCRIPTION="A free media library that is intended for use with a plex client."
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: We don't usually add a dot in the end of descriptions, just a minor styling thing.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+HOMEPAGE="https://www.plex.tv/"
+SRC_URI="amd64? ( ${URI}/${_FULL_VERSION}/debian/plexmediaserver_${_FULL_VERSION}_amd64.deb )"
+SLOT="0"
+LICENSE="Plex"
+RESTRICT="bindist strip"
+KEYWORDS="-* ~amd64"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+DEPEND="
+	sys-apps/fix-gnustack
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: I believe this is a BDEPEND. Could you verify? 
DEPEND is needed for libraries and headers, BDEPEND only when something is executed during building.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+S="${WORKDIR}"
+PATCHES=( "${FILESDIR}/virtualenv_start_pms_2019.patch" )
+
+src_unpack() {
+	unpack_deb ${A}
+}
+
+src_install() {
+	# Move the config to the correct place
+	local CONFIG_VANILLA="/etc/default/plexmediaserver"
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: local variables are written with smallcase lettering, to differentiate from GLOBAL ones which shouldn't be modified within these phases.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+	local CONFIG_PATH="/etc/${_SHORTNAME}"
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: Similarily here.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+	dodir "${CONFIG_PATH}"
+	insinto "${CONFIG_PATH}"
+	doins "${CONFIG_VANILLA#/}"
+	sed -e "s#${CONFIG_VANILLA}#${CONFIG_PATH}/${_APPNAME}#g" -i "${S}"/usr/sbin/start_pms || die
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: Actually this all looks a bit stupid and can be made in 1-2 lines.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+	# Remove Debian specific files
+	rm -rf "usr/share/doc" || die
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: No `-f`, you need to know what you're removing and when ;)

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+
+	# Copy main files over to image and preserve permissions so it is portable
+	cp -rp usr/ "${ED}"/ || die
+
+	# Make sure the logging directory is created
+	local LOGGING_DIR="/var/log/pms"
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: local...

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+	dodir "${LOGGING_DIR}"
+	chown "${_USERNAME}":"${_USERNAME}" "${ED}/${LOGGING_DIR}" || die
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: `fowners` should work better, allows to remove `|| die` as well.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+	keepdir "${LOGGING_DIR}"
+
+	# Create default library folder with correct permissions
+	local DEFAULT_LIBRARY_DIR="/var/lib/${_APPNAME}"
+	dodir "${DEFAULT_LIBRARY_DIR}"
+	chown "${_USERNAME}":"${_USERNAME}" "${ED}/${DEFAULT_LIBRARY_DIR}" || die
+	keepdir "${DEFAULT_LIBRARY_DIR}"
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: Now I am really not sure whether to trust acct- with this one or not, it should create the DIR with correct permissions already. This is the duplication that devmanual warns about.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+	# Install the OpenRC init/conf files
+	doinitd "${FILESDIR}/init.d/${PN}"
+	doconfd "${FILESDIR}/conf.d/${PN}"
+
+	# Disabling due to Bug 644694
+	#_handle_multilib
+
+	# Mask Plex libraries so that revdep-rebuild doesn't try to rebuild them.
+	# Plex has its own precompiled libraries.
+	_mask_plex_libraries_revdep
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: I wonder why this has to be its own function? Wouldnt it be cleaner to just handle it here?

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+
+	# Install systemd service file
+	local INIT_NAME="${PN}.service"
+	local INIT="${FILESDIR}/systemd/${INIT_NAME}"
+	systemd_newunit "${INIT}" "${INIT_NAME}"
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: Can be squished to one line...

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+
+	_remove_execstack_markings
+	_add_pax_markings
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: These too could be just extracted here.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+# Finds out where the library directory is for this system
+# and handles ldflags as to not break library dependencies
+# during rebuilds.
+_handle_multilib() {
+	# Prevent revdep-rebuild, @preserved-rebuild breakage
+	cat > "${T}"/66plex <<-EOF || die
+		LDPATH="${EPREFIX}/usr/$(get_libdir)/plexmediaserver"
+	EOF
+
+	doenvd "${T}"/66plex
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: `newenvd` can install this without having to resort to a temporary directory, nor having to call `|| die` at all.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+	# then so should we.
+	echo "SEARCH_DIRS_MASK=\"${EPREFIX}/usr/lib/plexmediaserver\"" > "${ED}"/etc/revdep-rebuild/80plexmediaserver
+}
+
+# Remove execstack flags from some libraries/executables
+# so that it works in hardened setups.
+_remove_execstack_markings() {
+	for f in "${EXECSTACKED_BINS[@]}"; do
+		# Unquoting 'f' so that expansion works.
+		fix-gnustack -f ${f} > /dev/null
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: `|| die`

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild (hunk 3 above)
Comment (ss23) [reply media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: Well, we can remove it from Gentoo, but upstream only supports 2.7, so I'm not sure what the approach should be. As the software isn't opensource, I don't think there's much we can do.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild (hunk 8 above)
Comment (ss23) [reply media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: I don't know what you mean by this. Is there a function that will do that all at once?

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild (hunk 17 above)
Comment (ss23) [reply media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: Removed as the file that was meant to be marked doesn't exist in this version of Plex.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild (hunk 16 above)
Comment (ss23) [reply media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: That entire function was never called due to it breaking systems, so I removed it.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild (hunk 12 above)
Comment (ss23) [reply media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: I've left it there for now, but this is something I'm going to revisit in later versions of the ebuild (as mentioned, this ebuild is still 2 major versions behind, so I'd like to get this one merged and start on a newer one ASAP).

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild (hunk 8 above)
Comment (juippis) [reply media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: I mean, I don't think using these paths twice is important enough to make them into a new variable, but up yours to decide. I'd find it much simpler if I saw the paths in `sed` and `insinto` plainly.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild (hunk 12 above)
Comment (juippis) [reply media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: Actually, is the dir needed by the **package**, or the **user**? That's the difference.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild (hunk 3 above)
Comment (juippis) [reply media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: I was afraid so. What I basically meant was, if you see patches for python3 support or if you find that other distros support python3 for this, implement that in Gentoo as well :P but yes, won't ask to do that now. Just something to keep an eye for.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+
+DEPEND="dev-python/virtualenv[${PYTHON_USEDEP}]"
+
+RDEPEND="
+	net-dns/avahi
+	acct-user/plex
+	acct-group/plex
+	${PYTHON_DEPS}"
+
+BDEPEND="sys-apps/fix-gnustack"
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: I think you removed the function that calls this tool, so you can remove the whole BDEPEND as well.

File: media-tv/plex-media-server/plex-media-server-1.16.4.ebuild [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+	einfo "Configuring virtualenv"
+	virtualenv -v --no-pip --no-setuptools --no-wheel "${ED}"/usr/lib/plexmediaserver/Resources/Python || die
+	pushd "${ED}"/usr/lib/plexmediaserver/Resources/Python &>/dev/null || die
+	find . -type f -exec sed -i -e "s#${D}##g" {} + || die
+	popd &>/dev/null || die
+}
+
+pkg_postinst() {
+	einfo ""
+	elog "Plex Media Server is now installed. Please check the configuration file in /etc/${_SHORTNAME}/${_APPNAME} to verify the default settings."
```
Comment (juippis) [inline media-tv/plex-media-server/plex-media-server-1.16.4.ebuild]: Forgot to mention this the last time, but you should really wrap this `elog` into shorter lines. 
https://devmanual.gentoo.org/ebuild-writing/messages/index.html So 75 characters per line maximum.

**Comment by juippis** [comment]
Well, actually, can you split this into 3 commits where acct-group and acct-user ebuilds are in their own. Then if you wish you could fix the desktop QA notes, 
```diff
--- plex-media-server-1.16.4.ebuild	2019-12-13 17:36:12.271684926 -0000
+++ plex-media-server-1.16.4-r1.ebuild	2019-12-13 17:33:46.100432334 -0000
@@ -32,7 +32,6 @@
 	acct-group/plex
 	${PYTHON_DEPS}"
 
-QA_DESKTOP_FILE="usr/share/applications/plexmediamanager.desktop"
 QA_PREBUILT="*"
 QA_MULTILIB_PATHS=(
 	"usr/lib/${_APPNAME}/.*"
@@ -60,6 +59,10 @@
 	# Remove Debian specific files
 	rm -r "usr/share/doc" || die
 
+	# Fix QA warning about .desktop file.
+	sed -i 's|Audio;Music;Video;Player;Media;|AudioVideo;Music;Player;|g' \
+		usr/share/applications/plexmediaserver.desktop || die
+
 	# Copy main files over to image and preserve permissions so it is portable
 	cp -rp usr/ "${ED}"/ || die
 
@@ -96,6 +99,7 @@
 	pushd "${ED}"/usr/lib/plexmediaserver/Resources/Python &>/dev/null || die
 	find . -type f -exec sed -i -e "s#${D}##g" {} + || die
 	popd &>/dev/null || die
+
 }
 
 pkg_postinst() {
```

**Pull request closed without merging**